backends/ebpf/tests/ptf/checksum.py

```diff
@@ -29,3 +29,36 @@ def runTest(self):
         exp_pkt = Ether() / bytes.fromhex('313233343536373839 BB3D')
         testutils.send_packet(self, PORT0, pkt)
         testutils.verify_packet_any_port(self, exp_pkt, ALL_PORTS)
+
+
+class InternetChecksumPSATest(P4EbpfTest):
+    """
+    Test if checksum in IP header (or any other using Ones Complement algorithm)
+    is computed correctly.
+    1. Generate IP packet with random values in header.
+    2. Verify that packet is forwarded. Data plane will decrement TTL twice and change
+     source IP address.
+    """
+
+    p4_file_path = "p4testdata/internet-checksum.p4"
+
+    def random_ip(self):
+        return ".".join(str(random.randint(0, 255)) for _ in range(4))
+
+    def runTest(self):
+        for _ in range(1):
+            # test checksum computation
+            pkt = testutils.simple_udp_packet(pktlen=random.randint(100, 512),
```

Comment:
in general we don't like random numbers since they make bugs harder to reproduce. But the coverage may be better; I am not sure what to say.

Reply:
Perhaps log the random number on failure

Reply:
PTF test prints whole packets on failure and leaves *.pcaps so I think it's enough.